GillianCore/debugging/debugger/debugger.ml

```diff
@@ -650,27 +686,46 @@ struct
     mutable lifted_exec_map : branch_case ExecMap.t option;
     mutable proc_name : string option;
     mutable unify_maps : (rid * UnifyMap.t) list;
+    report_state : L.report_state;
+  }
+
+  type debug_cfg = {
+    source_file : string;
+    source_files : SourceFiles.t option;
+    prog : Verification.prog_t;
+    tl_ast : tl_ast option;
+    tests : (string * Verification.t) list;
+    main_proc_name : string;
+    report_state_base : L.report_state;
   }
 
+  type debug_state = {
+    cfg : debug_cfg;
+    procs : (string, debug_proc_state) Hashtbl.t;
+    mutable cur_proc_name : string;
+  }
+
+  let get_proc_state ?proc_name ?(activate_report_state = true) dbg =
```

Comment:
You added `?proc_name` to a lot of functions, but it seems that you're never using the parameter.
Regarding optional parameters, I prefer to add them as needed. They introduce new control flows, which is always nice to not do if possible

Reply:
Resolved in c0f957eaf54d88a6a982d9dd569a5cc9a30f4d56